README.md

```diff
@@ -3,34 +3,51 @@
-- managing Storage Volumes (Persistent Volumes) and 
-- managing Storage Infrastructure (Physical Storage - Local/Cloud Disks/SSDs/Cache, Controllers, Networks). 
+OpenEBS Maya extends the capabilities of Kubernetes to orchestrate CAS (aka Container Native) Storage Solutions like OpenEBS Jiva, OpenEBS cStor, etc. *Maya* (meaning *Magic*), seamlessly integrates into the Kubernetes Storage Workflow and helps provision and manages the CAS based Storage Volumes. The core-features of *Maya* include:
+- Maintaining the inventory of the underlying disks on the Kubernetes Nodes 
+- Managing the allocation of Disks to CAS Storage Engines
+- Provisioning of CAS Volumes by interfacing with K8s Dynamic Volume Provisioner
+- Managing the high availability of the CAS volumes by tuning the scheduling parameters of CAS Deployments (Pods)
+- Provide adapters to CAS Volumes to interact with Kubernetes and related infrastructure components like Prometheus, Grafana etc.
 
-OpenEBS Maya allows you to manage storage across multiple zones (aka clusters/environments), that are co-located or geographically separated and can also run from within a single host. Maya can move the storage across different tiers based on the application needs (volume migration). Maya can learn and adapt to the changing environment through machine learning and data analytics. 
+*Maya* orchestration and management capabilities are delivered through a set of services and tools. Currently these services support deploying the CAS Storage Solutions in Kubernetes Clusters. In the future, these can be extended to support other Container Orchestrators.
 
-## Design
-
-Maya - *Storage Orchestration* functionality is delivered through a set of services and tools that seamlessly integrate *Maya* into your Container Orchestrators like Kubernetes, Docker Swarm, etc.,  Maya comprises of several components, which are themself delivered as container images. 
+## Maya Architecture 
+![](./docs/openebs-maya-architecture.png)
 
 **Maya** components can be broadly classified based on their deployment type as follows:
 
-- **Cluster Components** - like API Server (maya-apiserver) that helps in processing the requests for creating new OpenEBS Volumes. The API server, will use the container orchestrators to provision the OpenEBS Volume Containers. Another example is the Smart Analytics (maya-mulebot) engine that gathers the data via the machine learning probes and runs heuristics analysis to optimize storage deployment. 
+- **Control Plane Components** - These are containers that are initialized as part of enabling the OpenEBS in a Kubernetes cluster.
+  * *maya-apiserver* helps with creation of CAS Volumes and provides API endpoints to manage those volumes. *maya-apiserver* can also be considered as an template engine that can be easily extended to support any kind of CAS storage solutions. It takes as input a set of CAS Templates that are converted into CAS K8s YAMLs based on the user requests.
```

Comment:
as an template = as a template
CAS Templates = CAS templates
on the user = on user

Reply:
done